content/docs/reference/tutorials/kubernetes/tutorial-migrate-nodegroups.md

```diff
@@ -0,0 +1,454 @@
+
+- [Deployments][k8s-deployments]
+- [PodDisruptionBudget][k8s-pdb]
+- [Node Selectors & Affinity][k8s-node-affinity]
+- [Pod Affinity & Anti-Affinity][k8s-pod-affinity]
+- [Service Selectors][k8s-service]
+- [Readiness & Liveness Probes][k8s-probes]
+- [Lifecycle Hooks][k8s-hooks]
+- [Termation Grace Period & Graceful Shutdown][k8s-graceful-shutdown]
+- [Pod Termination Lifecycle][pod-lifecycle]
+- [Zero-downtime Deployment in Kubernetes with Jenkins][k8s-jenkins]
+
+## The Great Migration
+
+Upon initial deployment of our workload stack, the `echoserver` will ultimately land
+on the Standard node group we've created, given our lack of specificity
+for where the Kubernetes scheduler should place it. The NGINX Ingress Controller
+instead, specifically targets the `2xlarge` node group given its heftier specs.
+Both scheduling choices were made with the intent of segmenting our workloads
+by use-case, and performance requirements we've identified.
+
+In our [inital update][initial-update] we selected that our EKS 
+control plane run on `v1.13.7` of Kubernetes, that the Standard node group
+use `v1.13.7` of the [EKS-optimized][eks-amis] worker AMIs, and that the
+`2xlarge` node group use `v1.12.7` workers.
+
+Since we do not want our `2xlarge` workers to drift too far apart from the control
+plane's version to cause [unsupported skew][version-skew], and we
+ultimately realize that the `2xlarge` node group may not suffice for a
+large, anticipated request load, we've decided to update various settings of 
+the node group for NGINX.
+
+The node group that NGINX will select and target will go from:
+
+  * Using `v1.12.7`of Kubernetes, in a pool of (3) `t3.2xlarge` worker instances ->
+  * Using `v1.13.7`of Kubernetes, in a pool of (5) `c5.4xlarge` worker instances.
+
+We will also be using the [`hey`][hey-load-testing] load testing tool through
+out our migration to consistently hit the endpoint and path for the
+`echoserver` behind NGINX to ensure we acheive zero-downtime.
+
+### Step 0: Launch Load Tests
+
+As we migrate NGINX from the `2xlarge` -> `4xlarge` node group, we'll kick off
+a load testing script against the endpoint and path of the `echoserver` on our
+cluster.
+
+You can install the [`hey`][hey-load-testing] load testing tool locally to your
+machine by doing the following:
+
+```bash
+go get -u github.com/rakyll/hey
+```
+
+Using the `LB` environment variable previously defined in the 
+`pulumi stack output` for NGINX's AWS load balancer, and a helper script in
+`./scripts`, we'll load test for many iterations of 50,000 requests, with 100
+concurrent requests at a time, e.g. run testing across 75 iterations:
+
+```bash
+./scripts/load-testing.sh $LB 75
+```
+
+> **Note**: Given the large number of requests being generated during load
+> testing (millions), a seperate machine for testing would be best suited
+for overall network performance and throughput on your client.
+
+### Step 1: Creating the new `4xlarge` Node Group
```

Comment:
I noticed this in many of our existing content, but we should probably start staying away from gerunds.

Reply:
Done